File: tests/fixtures/cpp/edge.cpp
// ventouse C++ fixtures — edges (P1 surface).
#include <stdexcept>
#include <vector>

// CPP-EC-THROW — `throw` is control flow, NOT an effect (P1) -> clean 0.
int checked(int x) {
    if (x < 0) {
        throw std::runtime_error("neg");
    }
    return x;
}

// CPP-EC-STATICMUT — writing a mutable static member -> dirty 20 ; class 10.
struct Counter {
    static int total;
    void add() { total += 1; } // writes a mutable static -> dirty 20
};

// CPP-EC-LOCAL — a local owned vector mutated locally -> clean 0.
int build(int n) {
    std::vector<int> v;
    for (int i = 0; i < n; i++) { v.push_back(i); }
    return v.size();
}

File: tests/fixtures/ts/edge.ts
// ventouse TS fixtures — edges (P1/P2 surface).

// TS-EC-THROW — `throw` is control flow, NOT an effect (P1) -> clean 0.
function checked(x: number): number {
  if (x < 0) {
    throw new Error("neg");
  }
  return x;
}

// TS-EC-ENUM — an `enum` is a runtime object declaration with no effect -> module clean.
enum Color {
  Red,
  Green,
}

// TS-EC-ASCONST — `as const` is an immutability hint; a module-level literal has no effect -> clean.
const CONFIG = { retries: 3 } as const;

// TS-EC-DECORATOR — a decorator is ignored for purity (the decorator call is not counted).
function sealed(_target: unknown): void {}

@sealed
class Service {
  handle(): number {
    return 1; // pure -> clean 0
  }
}
